File: 141-Linked-List-Cycle/py/main.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        if not head or not head.next :
            return False
        
        seen = set()
        current = head 

        while current :
            if current in seen :
                return True
            seen.add(current)
            current = current.next
        return False

File: 141-Linked-List-Cycle/ts/main.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function hasCycle(head: ListNode | null): boolean {
    if (!head || !head.next ) {
        return false;
    }

    let seen = [];
    let current = head

    while (current != null) {
        if (seen.includes(current)) {
            return true;
        }
        seen.push(current)
        current = current.next
    }
    return false
    
};
